core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,14 +32,14 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).success(function() {
+        $http.put(collectionRightsUrl, putParams).then(function() {
           // TODO(bhenning): Consolidate the backend rights domain objects and
           // implement a frontend activity rights domain object. The rights
           // being passed in here should be used to create one of those objects.
           if (successCallback) {
             successCallback();
           }
-        }).error(function(error) {
+        }, function(error) {
```

Comment:
Should this be errorResponse, and the stuff below be errorResponse.data? Please manually test.